filterprotobuffer.go

```diff
@@ -0,0 +1,55 @@
+package dapper
+
+import (
+	"bytes"
+	"github.com/dogmatiq/iago/indent"
+	"github.com/dogmatiq/iago/must"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/proto"
+	"io"
+	"reflect"
+)
+
+// protobufMessageType is the reflect.Type for proto.Message
+var protobufMessageType = reflect.TypeOf((*proto.Message)(nil)).Elem()
+
+// ProtobufferFilter is a filter for proto.Message types. It shows the field names as defined in the .proto file
+// and hides implementation-specific internal state.
+func ProtobufferFilter(
+	w io.Writer,
+	v Value,
+	c Config,
+	p FilterPrinter,
+) (err error) {
+	if !v.DynamicType.Implements(protobufMessageType) {
+		return
+	}
+
+	must.WriteString(w, p.FormatTypeName(v))
+	must.WriteByte(w, '{')
+	defer must.WriteByte(w, '}')
+
+	message := v.Value.Interface().(proto.Message)
+
+	// prototext.MarshalOptions only allows tabs and spaces to be used for indentation.
```

Comment:
Out of curiosity what does it do if the indent string contains other characters?

Reply:
Depends on the method you use. If using Format() it fails silently. If using Marshal() it returns an error.